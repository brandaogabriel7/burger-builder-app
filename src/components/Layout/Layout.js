import React, { Fragment } from 'react';

const Layout = ({ children }) => (
    <Fragment>
        <div>Toolbar, SideDrawer, Backdrop</div>
        <main>{children}</main>
    </Fragment>
);

export default Layout;
