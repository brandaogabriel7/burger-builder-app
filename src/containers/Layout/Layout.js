import React, { Component, Fragment } from 'react';
import SideDrawer from '../../components/Navigation/SideDrawer/SideDrawer';
import Toolbar from '../../components/Navigation/Toolbar/Toolbar';
import styles from './layout.module.css';

class Layout extends Component {
    state = {
        showSideDrawer: false
    };

    sideDrawerToggledHandler = () => {
        this.setState(prevState => ({ showSideDrawer: !prevState.showSideDrawer }));
    };

    render() {
        return (
            <Fragment>
                <Toolbar drawerToggleClicked={this.sideDrawerToggledHandler} />
                <SideDrawer show={this.state.showSideDrawer} closed={this.sideDrawerToggledHandler} />
                <main className={styles.Content}>{this.props.children}</main>
            </Fragment>
        );
    }
}

export default Layout;
