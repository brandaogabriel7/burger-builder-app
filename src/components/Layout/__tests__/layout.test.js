import { render, screen } from '@testing-library/react';
import React from 'react';
import Layout from '../Layout';

const renderChildrenTestCases = [
    [<button>batata</button>, 'button', 'batata'],
    [<a href="https://www.google.com">google</a>, 'link', 'google'],
    [<h1>this is the title</h1>, 'heading', 'this is the title']
];

const matchSnapshotTestCases = [
    [
        <h1>batata</h1>,
        <ul>
            <li>a</li>
            <li>b</li>
            <li>c</li>
            <li>d</li>
            <li>e</li>
        </ul>
    ],
    [
        <form>
            <input type="text" />
            <button>Submit</button>
        </form>
    ]
];

describe('layout should', () => {
    it('render correctly', () => {
        render(<Layout />);
        expect(
            screen.getByText(/toolbar, sidedrawer, backdrop/i)
        ).toBeInTheDocument();
    });

    it.each(renderChildrenTestCases)(
        "render children '%j' correctly",
        (children, role, name) => {
            render(<Layout>{children}</Layout>);
            expect(screen.getByRole(role, { name })).toBeInTheDocument();
        }
    );

    it.each(matchSnapshotTestCases)('match snapshot for %j', (children) => {
        const container = render(<Layout>{children}</Layout>);
        expect(container).toMatchSnapshot();
    });
});
