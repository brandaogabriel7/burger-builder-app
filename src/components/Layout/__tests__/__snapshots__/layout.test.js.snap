// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout should match snapshot for {"type":"form","key":null,"ref":null,"props":{"children":[{"type":"input","key":null,"ref":null,"props":{"type":"text"},"_owner":null,"_store":{}},{"type":"button","key":null,"ref":null,"props":{"children":"Submit"},"_owner":null,"_store":{}}]},"_owner":null,"_store":{}} 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        Toolbar, SideDrawer, Backdrop
      </div>
      <main>
        <form>
          <input
            type="text"
          />
          <button>
            Submit
          </button>
        </form>
      </main>
    </div>
  </body>,
  "container": <div>
    <div>
      Toolbar, SideDrawer, Backdrop
    </div>
    <main>
      <form>
        <input
          type="text"
        />
        <button>
          Submit
        </button>
      </form>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`layout should match snapshot for {"type":"h1","key":null,"ref":null,"props":{"children":"batata"},"_owner":null,"_store":{}} 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        Toolbar, SideDrawer, Backdrop
      </div>
      <main>
        <h1>
          batata
        </h1>
      </main>
    </div>
  </body>,
  "container": <div>
    <div>
      Toolbar, SideDrawer, Backdrop
    </div>
    <main>
      <h1>
        batata
      </h1>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
